src/storage/src/table/storage_table.rs

```diff
@@ -154,6 +155,56 @@ impl<S: StateStore, RS: RowSerde> StorageTableBase<S, RS, READ_ONLY> {
             distribution,
         )
     }
+
+    pub fn from_table_catalog(
+        table_catalog: &Table,
+        store: S,
+        vnodes: Option<Arc<Bitmap>>,
+    ) -> Self {
+        let table_columns: Vec<ColumnDesc> = table_catalog
```

Comment:
May need to DRY this part.

Reply:
Let's call this in `StateTable::from_table_catalog` and create a state table with a default memory table?

Reply:
State Table seems only can create StorageTable<READ_WRITE> ?

Reply:
Otherwise I guess we would like to introduce this Read/Write concept in StateTable

Reply:
The state table should always be `READ_WRITE` since it's used by executors. How about `impl<S: StateStore, RS: RowSerde, const T: AccessType> StorageTableBase<S, RS, T>`?

Reply:
I think it is just the current way? Now use StorageTableBase::new_partial. 
```rust
impl<S: StateStore, RS: RowSerde> StorageTableBase<S, RS, READ_ONLY> {
    /// Create a read-only [`StorageTableBase`] given a complete set of `columns` and a partial
    /// set of `column_ids`. The output will only contains columns with the given ids in the same
    /// order.
    /// This is parameterized on cell based row serializer.
    pub fn new_partial(
        store: S,
        table_id: TableId,
....
```
What I originally want to do is to parse table columns, pk_indices stuff.

Reply:
We can implement `from_table_catalog` for storage table with generic access types. 🤔

Reply:
Fixed for you. 🥺

Reply:
mis-understand the idea. Thanks for the code!